GUIDELINES.md

```diff
@@ -0,0 +1,118 @@
+# Guidelines
+
+This document covers general best-practices when authoring Traits and/or
+Specifications for inclusion in the library.
+
+## Traits
+
+Traits define unique qualities or characteristics of an Entity or
+Relationship.
+
+- Traits should be atomic. No single trait should ever change the
+  meaning of another trait _or its properties_. This ensures that future
+  traits not known at the time code is written can change the meaning of
+  previously defined values.
+
+- Traits should define a single characteristic, consider splitting
+  'optional' properties into a separate trait with required properties.
```

Comment:
[consider] Worth mentioning that optional properties are allowed and supported by the API as null return values, but are  strongly discouraged?

Reply:
Tried to capture the offline discussion [here](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/71/commits/4c48d5a19fd4f992792b24b3df1b098cdedec890)